frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/CheckUserExistsHandler.java

```diff
@@ -167,8 +176,10 @@ public APIGatewayProxyResponseEvent handleRequestWithUserContext(
                                 .getValue();
 
                 LOG.info("Setting internal common subject identifier in user session");
-                userContext.getSession().setInternalCommonSubjectIdentifier(internalPairwiseId);
 
+                session.setInternalCommonSubjectIdentifier(internalPairwiseId);
+                maybeAuthSession.ifPresent(
```

Comment:
Hmm, I'm not sure about this. I think the account interventions lambda is generally next and it only uses this for auditing, so can fail gracefully, then in the login lambda it would get set anyway, but that will fail with a missing auth session, so we are just deferring the failure.

Reply:
Probably makes sense to just fail here then?